External/FEXCore/Source/Interface/Core/OpcodeDispatcher/Vector.cpp

```diff
@@ -990,6 +990,30 @@ void OpDispatchBuilder::VANDNOp(OpcodeArgs) {
   StoreResult(FPRClass, Op, Dest, -1);
 }
 
+template <size_t ElementSize>
+void OpDispatchBuilder::VHADDPOp(OpcodeArgs) {
+  const auto SrcSize = GetSrcSize(Op);
+  const auto Is256Bit = SrcSize == Core::CPUState::XMM_AVX_REG_SIZE;
+
+  OrderedNode *Src1 = LoadSource(FPRClass, Op, Op->Src[0], Op->Flags, -1);
+  OrderedNode *Src2 = LoadSource(FPRClass, Op, Op->Src[1], Op->Flags, -1);
+
+  OrderedNode *Res = _VFAddP(SrcSize, ElementSize, Src1, Src2);
+  OrderedNode *Dest = _VMov(SrcSize, Res);
```

Comment:
For the 256-bit case is this VMov necessary instead of using `Res` directly in the first SSA argument in the first `VInsElement` below?
Then only do VMov on the 128-bit case for the zero extension?

Reply:
Oh, good catch. Will fix that